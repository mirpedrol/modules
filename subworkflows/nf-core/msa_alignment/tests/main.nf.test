// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core subworkflows test msa_alignment
nextflow_workflow {

    name "Test Subworkflow MSA_ALIGNMENT"
    script "../main.nf"
    workflow "MSA_ALIGNMENT"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/msa_alignment"
    tag "mafft/align"
    tag "kalign/align"
    tag "famsa/align"
    tag "muscle5/super5"
    tag "magus/align"
    tag "clustalo/align"
    tag "tcoffee/align"
    tag "learnmsa/align"

    test("run mafft/align") {
        when {
            workflow {
                """
                input[0]  = Channel.of( [
                    [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/scaffolds.fasta', checkIfExists: true)
                        , 'mafft_align'
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match("mafft_align")}
            )
        }
    }

    test("run kalign/align") {

        when {
            workflow {
                """
                input[0]  = Channel.of( [ [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true)
                        , 'kalign_align'
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match("kalign_align")}
            )
        }
    }

    test("run famsa/align") {

        when {
            workflow {
                """
                input[0]  = Channel.of( [ [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true)
                        , 'famsa_align'
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.alignment).match("famsa_align_alignment")},
                { assert snapshot(workflow.out.versions).match("famsa_align_versions") }
            )
        }
    }

    test("run muscle5/super5") {

        when {
            workflow {
                    """
                        input[0]  = Channel.of( [ [ id:'test' ],
                        file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                        , 'muscle5_super5'
                    ])
                        """
            }
        }

            then {
                assertAll(
                        { assert workflow.success },
                        { assert snapshot(workflow.out.alignment).match("muscle5_super5_alignment") },
                        { assert snapshot(workflow.out.versions).match("muscle5_super5_versions") }
                        )
        }
    }

    test("run magus/align") {

        when {
            workflow {
                """
                input[0]  = Channel.of( [ [ id:'test' ], // meta map
                        file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                        , 'magus_align'
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                // tests seem to be reproducible on a single machine, but not across different machines
                // test the correct samples are in there
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1apf") },
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1ahl") },
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1atx") },
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1sh1") },
                { assert path(workflow.out.alignment[0][1]).getTextGzip().contains(">1bds") },
                { assert snapshot(workflow.out.versions).match("magus_align_versions") }
            )
        }
    }

    test("run clustalo/align") {
        tag 'this_one'

        when {
            workflow {
                """
                input[0]  = Channel.of( [ [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true)
                        , 'clustalo_align'
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match("clustalo_align")}
            )
        }
    }

    test("run tcoffee/align") {

        when {
            workflow {
                """
                input[0]  = Channel.of( [ [ id:'test' ],
                            file(params.modules_testdata_base_path + "../../multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                        , 'tcoffee_align'
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert path(workflow.out.alignment.get(0).get(1)).getTextGzip().contains("1ahl") },
                { assert snapshot(workflow.out.versions).match("tcoffee_align")}
            )
        }
    }

    test("run learnmsa/align") {
        options "-stub"
        config "./nextflow.config"

        when {
            workflow {
                """
                input[0]  = Channel.of( [ [ id:'test' ],
                        file(params.modules_testdata_base_path + "../../multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                        , 'learnmsa_align'
                    ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match("learnmsa_align") }
            )
        }
    }

}