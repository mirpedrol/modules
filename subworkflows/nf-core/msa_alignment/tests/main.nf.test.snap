{
    "kalign_align_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln:md5,c165ecf48fb89862cc2a991cc3cadb2d"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T14:14:36.278890895"
    },
    "learnmsa_align_versions": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T14:25:08.672798822"
    },
    "clustalo_align_alignment": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T14:16:39.69612288"
    },
    "kalign_align_versions": {
        "content": [
            [
                "versions.yml:md5,e7d33c95bb5d69e8573c8ad4eb2aca2c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T14:14:36.336612634"
    },
    "clustalo_align_versions": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T14:16:39.75436747"
    },
    "mafft_align_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.fas:md5,23426611f4a0df532b6708f072bd445b"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T14:50:14.382788"
    },
    "mafft_align_versions": {
        "content": [
            [
                "versions.yml:md5,4eac1bd98ea7280bd9822cc13631b619"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T14:50:14.431466"
    },
    "tcoffee_align_versions": {
        "content": [
            [
                "versions.yml:md5,c5208e86b43e8c973c39c2bd8ca2932a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T13:13:11.244144"
    },
    "tcoffee_align_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln:md5,a99aa6aa94e11653da64ff5c0c3a0d85"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T14:18:31.339885109"
    },
    "famsa_align_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln:md5,7cf7375f2ba360814ea978731838b972"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T13:11:35.22561"
    },
    "famsa_align_versions": {
        "content": [
            [
                "versions.yml:md5,c74ca8b91c442fc4ea29219ee1b724fd"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T13:11:35.27748"
    },
    "muscle5_super5_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln:md5,46ba556df08f7aabbe5e1ba31d226b6a"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T10:59:19.096521"
    },
    "magus_align_versions": {
        "content": [
            [
                "versions.yml:md5,9258e7c6deb7c3d816ba75cf111e09a8"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T13:12:53.134599"
    },
    "muscle5_super5_versions": {
        "content": [
            [
                "versions.yml:md5,48ec1b7cf99109e8495f3bc00d67a1eb"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T10:59:19.104182"
    }
}